packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -470,6 +470,13 @@ export class GraphqlApi extends GraphqlApiBase {
    */
   public readonly apiKey?: string;
 
+  /**
+   * the associated custom AppSync domain, if present
+   *
+   * @default - no custom associated domain
+   */
+  public readonly appSyncDomainName?: string;
+
```

Comment:
```suggestion
  /**
   * The associated domain name provided by AppSync
   */
  public readonly appSyncDomainName: string;

```
`domainName.attrAppSyncDomainName` seems to be always available. Is there a reason why you made this optional at first?

Reply:
The domainName resource is only created when you pass the domainName properties in the api. So when creating a minimal api it will not create an AWS::AppSync::DomainName or an AWS::AppSync::DomainNameApiAssociation and therefore you don't have the appSyncDomainName.

Reply:
Sorry I only just saw this comment appear. I have changed it a bit to be even more clear when you are requesting the appSyncDomainName without a domainName configuration.